qiskit/visualization/pulse_visualization.py

```diff
@@ -26,39 +28,85 @@
     from matplotlib import get_backend
 
 
-def pulse_drawer(data, dt=1, style=None, filename=None,
-                 interp_method=None, scale=None, channel_scales=None,
-                 channels_to_plot=None, plot_all=False, plot_range=None,
-                 interactive=False, table=True, label=False, framechange=True,
-                 channels=None, scaling=None,
-                 show_framechange_channels=True):
-    """Plot the interpolated envelope of pulse
+def pulse_drawer(data: Union['SamplePulse', 'ScheduleComponent'],
```

Comment:
To remove build warnings. (`WARNING: Cannot resolve forward reference in type annotations of "qiskit.visualization.pulse_drawer": name 'ScheduleComponent' is not defined`)

```suggestion
def pulse_drawer(data: Union[SamplePulse, ScheduleComponent],
```
Please import the necessary modules. Similarly, update `PulseStyle` and `SchedStyle`

Reply:
Type hinting in visualizer often cause cyclic import error. This time the error is very strange because `test/python/validation/__init__.py` never call visualization module. I'll check in detail this weekend.
https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=9107&view=logs&j=7684b0c9-af38-5400-184f-6880bd8a4dd7&t=9d95188b-4e59-58d8-f7c4-3f1fbd4209ad